go/vt/topo/test/trylock.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"context"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"vitess.io/vitess/go/vt/topo"
+
+	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
+)
+
+// checkLock checks we can lock / unlock as expected. It's using a keyspace
+// as the lock target.
+func checkTryLock(t *testing.T, ts *topo.Server) {
```

Comment:
Good tests. Just to be on the safe side (though it shouldn't matter), it might make sense to run some of these at keyspace level and some at shard level.

Reply:
The way I am reading the code , we would need to create semantics of TryCreateKeyspace etc. I think we can do that as we move towards these semantics in future.